docs/content/v0.20/release.md

```diff
@@ -0,0 +1,60 @@
+- There is a concept of a release branch, with a name following the pattern `release-$major.$minor`. This allows contributors to do maintenance work on older release branches. This also allows a user to track work along such a line.
+- The contents of `main` always work. This includes passing CI tests. This includes documentation being accurate. We allow point-in-time specific documentation, such as a document that says "Here is how to use release 1.2.3" --- which would refer to a release made in the past. We do not require the documentation in `main` to document all releases.
+- A git tag is immutable. Once associated with a given Git commit, that association is not changed later.
+- We do not put self-references into Git. For example, making release `1.2.3` doe not require changing any file in Git to have the string `1.2.3` in it.
+
+We have the following limitations.
+
+- The only way to publish artifacts (broadly construed, not (necessarily) GitHub "release artifacts") is to make a release.
+- The only way to test published artifacts is to make a release and test it.
+- Thus, it is necessary to keep users clearly appraised of the quality (or status of evaluating the quality) of each release.
+- Because of the lack of self references, most instructions and tests do not have concrete release identifiers in them; instead, the user has to chose and supply the release identifier. There can also be documentation of a specific past release (e.g., the latest stable release) that uses the literal identifier for that past release.
+- Because of the prohibition of self references, Git will not contain the exact bytes of our Helm chart definitions. Where a Helm chart states its own version or supplies a default value for a container image reference to an image built from the same release, the bytes in Git have a placeholder for that image's tag and the process of creating the published release artifacts fills in that placeholder. Think of this as being analogous to the linking done when building a binary executable file.
+
+## Technology
+
+There will be a GitHub workflow that creates the published artifacts for each Git tag whose name starts with "v". The rest of the tag name is required to be a semver release identifier. For each tag `v$version` the following published artifacts will be created.
+
+- The container image for the kubestellar-controller-manager (KCM), at `ghcr.io/kubestellar/kubestellar-controller-manager` (or should it be `ghcr.io/kubestellar/kubestellar/controller-manager` from the ks/ks repo? or just `ghcr.io/kubestellar/controller-manager`?). Image tag will be `$version`.
+- The Helm chart (for installing the KCM for a WDS), at `ghcr.io/kubestellar/kubestellar-controller-manager-chart` with version `$version` and Helm "appVersion" `$version`. The `values.yaml` of this chart version has a default value for the image reference to the KCM and that value is `ghcr.io/kubestellar/kubestellar-controller-manager:$version`.
+
```

Comment:
Should this section include a list of all the images/artifacts produced by the kubestellar organization, including kubeflex, status-addon, etc, and how they should be synchronized?

Reply:
For repos ks/ks depends on, "no". I think we can just be ordinary here, it repo takes care of itself.

Referring to https://github.com/kubestellar/kubestellar/blob/main/docs/content/v0.20/packaging.md#outline-of-publishing, and adding in the dashed where ks/ocm-status-addon depends on ks/ks (look in the instructions in ks/ocm-status-addon), that leaves two repos of concern: ks/ocm-transport-plugin and ks/ocm-status-addon. In some sense there is actually a cycle of dependency here, because you can not use ks/ks without also using those repos.

I think it would be cleanest to split ks/ks into two parts. Think of it as two layers. The lower layer holds the bulk of the functionality and does not depend on ks/ocm-anything. The upper layer brings it all together, depending on ks/ks, ks/ocm-status-addon, and ks/ocm-transport-plugin.